dts/bindings/i2s/st,stm32-i2s.yaml

```diff
@@ -25,3 +25,12 @@ properties:
 
   pinctrl-names:
     required: true
+
+  mck-enabled:
+    type: phandle
```

Comment:
Couldn't it be a boolean ?
You just need to know if a mck pin is enabled, right ?

Reply:
The reason why it isn't just a boolean is because a phandle allows to perform a verification step on the user selection, which overall improves the coherence on the enabling action in my opinion.

As an example, consider what the following scenario would yield if `mck-enabled` was a boolean but no suitable alternate function pin provided.

```
/* DT */
pinctrl-0 = <... &i2sX_mck_pin missing >;
mck-enabled;

/* I2S driver */
.master_clk_sel = DT_NODE_HAS_PROP(..., mck_enabled)
```
Admittedly, that would be incorrect usage as no mck_pin was provided, albeit not a far fetched possibility I believe.  One can imagine trying to debug such a case would be quite frustrating since `.master_clk_sel` would look set to the user. A phandle with a relevant check would be better aligned with the end result, still relies in the appropriate pin being listed for both properties (twice in total, so it is also reinforcing :) )

```
/* DT */
pinctrl-0 = <... &i2sX_mck_pYZ>;
mck-enabled = <&i2sX_mck_pYZ>;
```
Ultimately, I would really like to see an verbose error of some sort thrown under user misusage. However, I couldn't find a solution to cover for that. @gmarull alluded in his comment above, it would be interesting to hear any suggestions on it.